internal/internal_task_pollers.go

```diff
@@ -798,13 +799,25 @@ func (wtp *workflowTaskPoller) poll(ctx context.Context) (interface{}, error) {
 }
 
 func (wtp *workflowTaskPoller) toWorkflowTask(response *s.PollForDecisionTaskResponse) *workflowTask {
+	startEventID := response.GetStartedEventId()
+	nextEventID := response.GetNextEventId()
+	if nextEventID != 0 &&
+		startEventID != 0 &&
+		nextEventID-1 != startEventID {
+		wtp.logger.Warn("Invalid PollForDecisionTaskResponse, nextEventID doesn't match startedEventID",
+			zap.Int64("StartedEventID", startEventID),
+			zap.Int64("NextEventID", nextEventID),
+		)
+		wtp.metricsScope.Counter(metrics.DecisionPollInvalidCounter).Inc(1)
```

Comment:
Should it return error or stop processing the response if it is invalid?

Reply:
Yes, but for this release I would like to be extra careful and first validating if the assumption here is true. If we don't see any logs and metrics for invalid decision we can return error here.